cpp/src/interop/to_arrow.cu

```diff
@@ -306,11 +312,19 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<cudf::string_view>(
                               static_cast<std::size_t>(sview.chars_size(stream))},
     ar_mr,
     stream);
-  return std::make_shared<arrow::StringArray>(static_cast<int64_t>(input_view.size()),
-                                              offset_buffer,
-                                              data_buffer,
-                                              fetch_mask_buffer(input_view, ar_mr, stream),
-                                              static_cast<int64_t>(input_view.null_count()));
+  if (cudf::strings::detail::is_large_strings_enabled()) {
+    return std::make_shared<arrow::LargeStringArray>(static_cast<int64_t>(input_view.size()),
+                                                     offset_buffer,
+                                                     data_buffer,
+                                                     fetch_mask_buffer(input_view, ar_mr, stream),
+                                                     static_cast<int64_t>(input_view.null_count()));
+  } else {
+    return std::make_shared<arrow::StringArray>(static_cast<int64_t>(input_view.size()),
+                                                offset_buffer,
+                                                data_buffer,
+                                                fetch_mask_buffer(input_view, ar_mr, stream),
+                                                static_cast<int64_t>(input_view.null_count()));
+  }
```

Comment:
This looks incorrect to me. There is no reason to check the enabled setting. Simply check the offsets type.
```suggestion
  if (sview.offsets().type().id()==cudf::type_id::INT64) {
    return std::make_shared<arrow::LargeStringArray>(static_cast<int64_t>(input_view.size()),
                                                     offset_buffer,
                                                     data_buffer,
                                                     fetch_mask_buffer(input_view, ar_mr, stream),
                                                     static_cast<int64_t>(input_view.null_count()));
  } else {
    return std::make_shared<arrow::StringArray>(static_cast<int64_t>(input_view.size()),
                                                offset_buffer,
                                                data_buffer,
                                                fetch_mask_buffer(input_view, ar_mr, stream),
                                                static_cast<int64_t>(input_view.null_count()));
  }
```